src/Sentry/Platforms/Android/SentrySdk.cs

```diff
@@ -100,7 +100,10 @@ private static void InitSentryAndroidSdk(SentryOptions options)
                 }
             }
 
-            o.BeforeSend = new BeforeSendCallback(BeforeSendWrapper(options), options, o);
+            if (options.Android.SuppressSegfaults || options.Native.EnableBeforeSend || options.BeforeSendInternal != null)
+            {
+                o.BeforeSend = new BeforeSendCallback(BeforeSendWrapper(options), options, o);
+            }
```

Comment:
There's a small issue with the operator in this expression... it should be:
```csharp
if (options.Android.SuppressSegfaults || (options.Native.EnableBeforeSend && options.BeforeSendInternal != null))
```

And this could probably be refined even further to eliminate the wrapper when it's not necessary:
```suggestion
            if (options.Android.SuppressSegfaults)
            {
                o.BeforeSend = new BeforeSendCallback(BeforeSendWrapper(options), options, o);
            }
            else if (options.Native.EnableBeforeSend && options.BeforeSendInternal is { } beforeSend)
            {
                o.BeforeSend = beforeSend;
            }
```

Ideally this would go in it's own PR as it seems unrelated to the issue this PR is trying to address.